src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/PooledBuilderExtensions.cs

```diff
@@ -92,33 +92,4 @@ public static ImmutableArray<T> ToFlattenedImmutableArrayAndFree<T>(this ArrayBu
             builders.Free();
         }
     }
-
-    public static bool HasDuplicates<T>(this IReadOnlyList<T> builder)
-        => builder.HasDuplicates(static x => x);
-
-    public static bool HasDuplicates<T, U>(this IReadOnlyList<T> builder, Func<T, U> selector)
```

Comment:
>public static bool HasDuplicates<T, U>(this IReadOnlyList<T> builder, Func<T, U> selector) [](http://example.com/codeflow?start=4&length=90)

Does this mean everything not an IA goes through the IEnumerable implementation now?

Reply:
Yeah, my concern was the alloc for ArrayBuilder since usage of that indicates a potentially hot path, but it kind of looks like we were already allocating the enumerator by doing a foreach in this this code when we already have an IReadOnlyList.

Reply:
I just changed it from ArrayBuilder to IReadOnlyList in previous PR. Hence this follow up. I'll specialize for `IReadOnlyList<T>` to avoid enumerator alloc.